### PR TITLE
E3: Remove HistRange-Based unwind

### PR DESCRIPTION
**Pull request by Giulio2002**
I had to remove one test, because it was testing the "unwind to tx" which now is not a possibility anymore

I also simplified the retrieval of the `Commitment` so now we do not need `IteratePrefix` and `IteratePrefix2` anymore.

Lastly, and naturally, all tests related to the deleted functions were also removed

### CODE REVIEW

File: erigon-lib/state/domain_test.go [hunk 1]
```diff
@@ -2082,230 +1753,6 @@ func TestDomain_PruneProgress(t *testing.T) {
 	fmt.Printf("exitiig after %d iterations\n", i)
 }
 
-func TestDomain_Unwind(t *testing.T) {
-	db, d := testDbAndDomain(t, log.New())
-	defer d.Close()
```
Comment (awskii) [inline erigon-lib/state/domain_test.go:2087]: why remvoe this test? Domains still provide Unwind functionality, that was naive test to check correctness after several unwinds on same data

File: erigon-lib/state/domain.go [hunk 2]
```diff
@@ -1574,146 +1482,6 @@ func (dt *DomainRoTx) GetLatestFromFiles(key []byte) (v []byte, found bool, file
 	return dt.getFromFiles(key)
 }
 
-func (dt *DomainRoTx) IteratePrefix(roTx kv.Tx, prefix []byte, it func(k []byte, v []byte) error) error {
```
Comment (awskii) [inline erigon-lib/state/domain.go:1577]: we will use tx.Range queries instead?

Comment (Giulio2002) [reply erigon-lib/state/domain.go:1577]: we wont use anything, this is useless now

File: erigon-lib/state/domain_test.go (hunk 1 above)
Comment (Giulio2002) [reply erigon-lib/state/domain_test.go:2087]: Because it tests unwind to tx

Comment (awskii) [reply erigon-lib/state/domain_test.go:2087]: this unit allows to check if some regression made in one Domain without being affected by others

Comment (Giulio2002) [reply erigon-lib/state/domain_test.go:2087]: current unwind unwinds all domains so it is obsolete

**Review by awskii — APPROVED** [review]

**Pull request merged**